HIVE-26565: Revamp Apache Hive website

simhadri-g opened the pull request.

zabetak commented on themes/hive/layouts/partials/banner.html:
```diff
@@ -0,0 +1,39 @@
+        <canvas id="canvas"></canvas>
+        <div class="main-title">
+            <img src="{{.Site.BaseURL}}{{ .Site.Params.hiveLogo }}" width="550" height="300"  alt="Apache Software Foundation">
+            <h1 class="text-outline black-text">Apache  Hive v4</h1>
```
Please remove v4 from the main header. This information can get obsolete and changing the main header may interfere with search engines.

simhadri-g replied:
Done

zabetak commented on themes/hive/layouts/partials/explain.html:
```diff
@@ -0,0 +1,43 @@
+  under the License. -->
+
+<explain>
+    <div id="explain" data-termynal data-termynal data-ty-typeDelay="40" data-ty-lineDelay="70"><span data-ty="input">jdbc:hive2://localhost:10001/> explain select * from test;</span>
+
+        <span data-ty>+---------------------------------------------+</span>
+        <span data-ty>                     Explain                      </span></span>
+        <span data-ty>+---------------------------------------------+</span>
+        <span data-ty>STAGE DEPENDENCIES:                               </span>
+        <span data-ty>  Stage-0 is a root stage                         </span>
+        <span data-ty>STAGE PLANS:                                      </span>
+        <span data-ty>  Stage: Stage-0                                  </span>
+        <span data-ty>    Fetch Operator                                </span>
+        <span data-ty>      limit: -1                                   </span>
+        <span data-ty>      Processor Tree:                             </span>
+        <span data-ty>        TableScan                                 </span>
+        <span data-ty>          alias: test                        </span>
+        <span data-ty>          Statistics: Num rows: 16 Data size: 2880 Basic stats: COMPLETE Column stats: NONE</span>
+        <span data-ty>          Select Operator                         </span>
+        <span data-ty>            expressions: id (type: int), name (type: string), intid (type: int)</span>
+        <span data-ty>            outputColumnNames: _col0, _col1, _col2</span>
+        <span data-ty>            Statistics: Num rows: 16 Data size: 2880 Basic stats: COMPLETE Column stats: NONE</span>
+        <span data-ty>            ListSink                              </span>
+        <span data-ty>+---------------------------------------------+</span>
+    </div>
```
Since we are talking about CBO here it would be better to change the example to use `EXPLAIN CBO` and maybe use a less trivial query with 2-3 joins, aggregation, and filter. Also using a well-known datataset (such as TPC-H or TPC-DS) would make this more realistic.

simhadri-g replied:
I have updated this with a join query on tpcds tables.

simhadri-g commented on .github/workflows/gh-pages.yml:
```diff
@@ -0,0 +1,36 @@
+          publish_dir: ./public
+          publish_branch: asf-site
+          destination_dir: ./
+          keep_files: true
```
done

simhadri-g commented on README.md:
```diff
@@ -1,24 +1,55 @@
-This directory contains the code for the Apache Hive web site,
-[hive.apache.org](https://hive.apache.org/). The easiest way to build
-the site is to use docker to use a standard environment.
+This repository contains the documentation for Apache Hive. 
```
Updated

simhadri-g commented on themes/hive/layouts/partials/head.html:
```diff
@@ -0,0 +1,38 @@
+<!-- - distributed under the License is distributed on an "AS IS" BASIS,-->
+<!-- - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.-->
+<!-- - See the License for the specific language governing permissions and-->
+<!-- - limitations under the License.-->
```
updated

simhadri-g commented on content/general/PrivacyPolicy.md:
```diff
@@ -21,15 +22,18 @@ layout: default
+4. The pages you visit; and
+5. The addresses of pages from where you followed a link to our site.
 
 Part of this information is gathered using a tracking cookie set by the
 <a href="http://www.google.com/analytics/">Google Analytics</a>
```
yes

simhadri-g commented on doap_Hive.rdf:
```diff
@@ -1,58 +0,0 @@
-<?xml version="1.0"?>
```
I have restored the file.

simhadri-g commented on hcatalog_downloads.md:
```diff
@@ -1,43 +0,0 @@
----
-title: HCatalog Downloads
```
I have added this file back.

simhadri-g commented on themes/hive/layouts/_default/baseof.html:
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
```
done.

simhadri-g commented on themes/hive/static/css/termynal.css:
```diff
@@ -0,0 +1,91 @@
+/**
+ * termynal.js
```
Added MIT License header.
The content for the header was taken from here: https://opensource.org/licenses/MIT

simhadri-g commented on themes/hive/static/js/script.js:
```diff
@@ -0,0 +1,102 @@
+var canvas = document.getElementById("canvas"),
```
I wrote the script.
I  have updated it with the apache license header.

simhadri-g commented on themes/hive/static/js/termynal.js:
```diff
@@ -0,0 +1,203 @@
+/*
```
Yes, i have update the file.

simhadri-g commented on themes/hive/theme.toml:
```diff
@@ -0,0 +1,21 @@
+# theme.toml template for a Hugo theme
```
This was an auto-generated file when the project was initialized with hugo. I have update the values in this commit.

simhadri-g commented:
Hi @zabetak ,

Sorry for the delay.
I have addressed the review comments and resolved the merge conflicts.
Please have a look when you are free.

simhadri-g commented on config.toml:
```diff
@@ -0,0 +1,51 @@
+baseURL = 'https://simhadri-g.github.io/hive-site'
```
Updated.

simhadri-g commented:
> Thanks for putting this together @simhadri-g ! I did a pass and left some comments/questions; most of them minor.
> 
> Regarding the images included in this PR did you verify that they are all licensed appropriately?

Images used in workflow are all MIT licensed:

1. peaceiris/actions-hugo ->  MIT(https://github.com/peaceiris/actions-hugo/blob/main/LICENSE)
2. actions/checkout@v3 -> MIT (https://github.com/actions/checkout/blob/main/LICENSE)
3. peaceiris/actions-gh-pages@v3 -> MIT (https://github.com/peaceiris/actions-gh-pages/blob/main/LICENSE)

The other components used: 
4. Hugo -> Apache-2.0 license(https://github.com/gohugoio/hugo/blob/master/LICENSE) 
5. termynal.js -> MIT license(https://github.com/ines/termynal/blob/master/LICENSE)

simhadri-g commented:
Updated website deployed here: https://simhadri-g.github.io/hive-site-hugo/index.html

zabetak commented on .github/workflows/gh-pages.yml:
```diff
@@ -0,0 +1,36 @@
+name: GitHub Pages
```
Please add AL2 header.

simhadri-g replied:
Updated the file with the AL2 header.

zabetak commented on themes/hive/theme.toml:
```diff
@@ -0,0 +1,21 @@
+features = []
+min_version = "0.41.0"
+
+[author]
+  name = "simhadri-g"
+  homepage = ""
```
Author tags are usually not present in Apache code so not sure if we should/can include this.

simhadri-g replied:
Ok, I will change the name to "hive".

zabetak reviewed (approved):
I think it is now in good shape to be merged but haven't tested locally yet. Will do hopefully in the next week and then we can give a heads up in the dev list and then push it! Awesome work ;)

simhadri-g commented:
Thanks @zabetak and thanks again for reviewing! 

Will wait for you to test!

We also have a backup of the old hive-site here until the new site goes online to avoid any disruptions. 
https://apache.github.io/hive/

zabetak commented on content/community/mailingList.md:
```diff
@@ -1,7 +1,9 @@
 ---
-title: Mailing Lists
-layout: default
+title: "MailingList"
```
MailingList -> MailingLists

simhadri-g replied:
Fixed

zabetak commented on content/community/mailingList.md:
```diff
@@ -20,46 +22,49 @@ layout: default
   specific language governing permissions and limitations
   under the License. -->
 
+# Mailing List
```
List -> Lists

simhadri-g replied:
Fixed

zabetak commented on themes/hive/layouts/partials/menu.html:
```diff
@@ -0,0 +1,114 @@
+                            <li><a class="dropdown-item" href="{{ .Site.Params.navbar.commiter }}">Becoming A Committer</a></li>
+                            <li><a class="dropdown-item" href="{{ .Site.Params.navbar.contribute }}">How To Contribute</a></li>
+                            <li><a class="dropdown-item" href="{{ .Site.Params.navbar.resourcesForDev }}">Resources for Contributors</a></li>
+                            <li><a class="dropdown-item" href="{{.Site.BaseURL}}{{ .Site.Params.navbar.mailinglist }}">Mailing List</a></li>
```
Mailing List -> Mailing Lists

simhadri-g replied:
Fixed

zabetak commented on config.toml:
```diff
@@ -0,0 +1,52 @@
+
+[params.features]
+    acidTxn = "https://cwiki.apache.org/confluence/display/hive/hive+transactions"
+    hs2 = "https://cwiki.apache.org/confluence/display/hive/hiveserver2"
```
Broken link: replace with `https://cwiki.apache.org/confluence/display/Hive/HiveServer2+Overview`?

simhadri-g replied:
Done

zabetak commented on themes/hive/layouts/partials/features.html:
```diff
@@ -0,0 +1,170 @@
+            </div>
+        </div>
+
+        <div class="row divs">
+            <div class="col-md feature-border">
+                <h2 class="topic-text-style">Hive Data Compaction</h2>
+                <p>
+                    Query-based and MR-based data compactions are supported out-of-the-box.
+                </p>
+                <a href="{{ .Site.Params.features.compactions }}" class="icon-block">
+                    <button class="custom-button feature-button-style" role="button">Learn More</button>
+                </a>
+            </div>
+            <div class="col-md">
+                {{- partial "compaction.html" . -}}
+            </div>
+        </div>
```
This is only relevant (and necessary) for ACID so maybe it is better to merge it with the previous section instead of highlighting as a key feature.

saihemanth-cloudera replied:
Was this comment addressed?

simhadri-g replied:
Not yet,  I would like to include it in a later patch along with a few other additions mentioned here: https://lists.apache.org/thread/f6f7h4w11bcggom94sovnrm3d22nwb43

zabetak commented on config.toml:
```diff
@@ -0,0 +1,52 @@
+    apacheEvents = "https://www.apache.org/events/current-event"
+    security = "https://www.apache.org/security/"
+    sponsors = "https://www.apache.org/foundation/thanks.html"
+    donate = "https://www.apache.org/foundation/sponsorship.html"
```
`donate = https://www.apache.org/foundation/contributing.html`

zabetak commented on config.toml:
```diff
@@ -0,0 +1,52 @@
+    privacyPolicy = "https://privacy.apache.org/policies/privacy-policy-public.html"
+    apacheEvents = "https://www.apache.org/events/current-event"
+    security = "https://www.apache.org/security/"
+    sponsors = "https://www.apache.org/foundation/thanks.html"
```
`sponsors = "https://www.apache.org/foundation/sponsorship.html"`

saihemanth-cloudera commented:
The mailing list button on the landing page is pointing to the community dev page.

saihemanth-cloudera commented on README.md:
````diff
@@ -1,24 +1,72 @@
+* Install [hugo]:
 
-## Pushing to site
+ ```brew install hugo```
````
Should we mention the installation procedure for different OS like windows/linux?

simhadri-g replied:
I have only tested out in MacOS. 
For other OS I have added a link to hugo installation page https://gohugo.io/installation/

saihemanth-cloudera commented on config.toml:
```diff
@@ -0,0 +1,52 @@
+    security = "https://www.apache.org/security/"
+    sponsors = "https://www.apache.org/foundation/thanks.html"
+    donate = "https://www.apache.org/foundation/sponsorship.html"
+    copyright = "The contents of this website are © 2022 Apache Software Foundation under the terms of the Apache License v2. Apache Hive and its logo are trademarks of the Apache Software Foundation."
```
Should this be 2023?

simhadri-g replied:
Updated the year.

saihemanth-cloudera commented on config.toml:
```diff
@@ -0,0 +1,52 @@
+theme = 'hive'
+[params]
+  hiveLogo = "/images/hive.svg"
+  logo = "/images/asf_logo.png"
```
Ideally, the logo should be clickable. You can hide the link 'Apache Hive' in the logo, so that it becomes clickable. You can increase the size of the logo a bit if you can.

simhadri-g replied:
Done

saihemanth-cloudera reviewed (changes requested):
The logo should be clickable. Left a few other comments as well.

simhadri-g commented on content/community/issueTracking.md:
```diff
@@ -21,11 +22,14 @@ layout: default
-  * Search the JIRA database.
-  * Check the [user mailing list](/mailing_lists.html), both by searching the archives and by asking questions.
+* Search the JIRA database.
+* Check the [user mailing list](/mailing_lists.html), both by searching the archives and by asking questions.
```
Done

simhadri-g commented:
Thanks @zabetak and @saihemanth-cloudera  for the reviews, I have updated the PR to address the review comments.
Updated site is deployed here: https://simhadri-g.github.io/hive-site-hugo/

zabetak commented:
Let's wait for @saihemanth-cloudera to approve the latest changes and then we can proceed with the merge.

saihemanth-cloudera commented on themes/hive/layouts/partials/compaction.html:
```diff
@@ -0,0 +1,28 @@
+        <span data-ty>Done!</span>
+        <span data-ty="input">jdbc:hive2://.../default> SHOW COMPACTIONS;</span>
+    </div>
+<!--    <script  src="{{ .Site.BaseURL }}/js/termynal.js" data-termynal-container="#comp"></script>-->
```
Nit:  remove this commented line.

simhadri-g replied:
Done.

saihemanth-cloudera commented on themes/hive/layouts/partials/footer.html:
```diff
@@ -0,0 +1,55 @@
+    </div>
+
+</footer>
+<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-MrcW6ZMFYlzcLA8Nl+NtUVF0sA7MsXsP1UyJoMp4YLEuNSfAP+JcXn/tWtIaxVXM" crossorigin="anonymous"></script>
```
Is it safe to use which encryption algorithm is used in the HTML?

simhadri-g replied:
Yes, it is safe. 
Bootstrap is included via CDN from the official source mentioned here: https://getbootstrap.com/ 
It looks like it is encrypted using SHA-384 which belongs to the same family as SHA-2. 
SHA-384 is supposed to be a faster hashing algorithm on 64-bit machines.

saihemanth-cloudera commented on themes/hive/layouts/partials/head.html:
```diff
@@ -0,0 +1,41 @@
+        <link rel="icon" href="/images/hive.svg" sizes="any" type="image/svg+xml">
+        <link rel="stylesheet" href="{{ .Site.BaseURL }}/css/hive-theme.css" />
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css">
+        <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">
```
Same as above comment.

simhadri-g replied:
Yes, it is safe. 
Bootstrap is included via CDN from the official source mentioned here: https://getbootstrap.com/ 
It looks like it is encrypted using SHA-384 which belongs to the same family as SHA-2. 
SHA-384 is supposed to be a faster hashing algorithm on 64-bit machines.

saihemanth-cloudera reviewed (commented):
Minor nits. Looks good otherwise.
